Matlab/Matlab.sublime-syntax

```diff
@@ -48,477 +76,792 @@ contexts:
+              scope: variable.parameter.attribute.matlab
+            - match: \b(?:public|protected|private)\b
+              scope: storage.modifier.matlab
+            - include: expressions
+
+  class-methods-body:
+    - meta_scope: meta.methods.matlab
+    - match: \bend\b
+      scope: keyword.control.end.methods.matlab
+      pop: true
+    - include: comments
+    - include: function-declaration
+
+  class-events:
+    - match: \bevents\b
+      scope: keyword.declaration.events.matlab
+      push:
+        - meta_scope: meta.events.matlab
+        - match: '{{eol}}'
+          set: class-events-body
+        - match: \(
+          scope: punctuation.section.parens.begin.matlab
+          push:
+            - meta_scope: meta.attributes.matlab
+            - include: line-continuation
+            - include: eol-pop
+            - match: \)
+              scope: punctuation.section.parens.end.matlab
+              pop: true
+            # https://www.mathworks.com/help/matlab/matlab_oop/event-attributes.html
+            - match: \b(?:Hidden|ListenAccess|NotifyAccess)\b
+              scope: variable.parameter.attribute.matlab
+            - match: \b(?:public|protected|private)\b
+              scope: storage.modifier.matlab
+            - include: expressions
+
+  class-events-body:
+    - meta_scope: meta.events.matlab
+    - match: \bend\b
+      scope: keyword.control.end.events.matlab
+      pop: true
+    - include: comments
+    - include: variable
+
+  class-enumeration:
+    - match: \benumeration\b
+      scope: keyword.declaration.enumeration.matlab
+      push: class-enumeration-body
+
+  class-enumeration-body:
+    - meta_scope: meta.enumeration.matlab
+    - match: \bend\b
+      scope: keyword.control.end.enumeration.matlab
+      pop: true
+    - include: expressions
+
+###[ KEYWORDS AND FLOW CONTROL BLOCKS ]#######################################
+
+  # https://www.mathworks.com/help/matlab/ref/iskeyword.html
+  # https://www.mathworks.com/help/matlab/control-flow.html
+  keywords:
+    - include: if
+    - include: switch
+    - include: for
+    - include: parfor
+    - include: while
+    - include: try
+    - include: spmd
+    - match: \bbreak\b
+      scope: keyword.control.flow.break.matlab
+    - match: \bcontinue\b
+      scope: keyword.control.flow.continue.matlab
+    - match: \breturn\b
+      scope: keyword.control.flow.return.matlab
+    - match: \bglobal\b
+      scope: storage.modifier.global.matlab
+    - match: \bpersistent\b
+      scope: storage.modifier.persistent.matlab
+
+  if:
+    - match: \bif\b
+      scope: keyword.control.conditional.if.matlab
+      push:
+        - meta_scope: meta.block.if.matlab
+        - match: \belseif\b
+          scope: keyword.control.conditional.elseif.matlab
+        - match: \belse\b
+          scope: keyword.control.conditional.else.matlab
+        - match: \bend\b
+          scope: keyword.control.end.if.matlab
```

Comment:
With regards to the former comment about `end` this is an example of what we used `keyword.control.conditional.end` for in some other languages.